include/nbl/asset/utils/CHLSLCompiler.h

```diff
@@ -51,8 +61,31 @@ class NBL_API2 CHLSLCompiler final : public IShaderCompiler
 				ret.setCommonData(options);
 			return ret;
 		}
+
+		class DxcCompilationResult
```

Comment:
You could move this and the function below it to the cpp only, or at least do the impl in the cpp